src/Bridge/Symfony/Validator/Validator.php

```diff
@@ -30,11 +31,13 @@ class Validator implements ValidatorInterface
 {
     private $validator;
     private $container;
+    private $serviceContainer;
 
-    public function __construct(SymfonyValidatorInterface $validator, ContainerInterface $container = null)
+    public function __construct(SymfonyValidatorInterface $validator, ContainerInterface $container = null, ContainerInterface $serviceContainer = null)
```

Comment:
No need to introduce a new argument here. We can just switch from the full container to the service locator in 3.0, since we trigger a deprecation error for services not implementing the interface.

Reply:
You mean just mark the generators as public to be available in the full container to avoid another container until 3.0?

Reply:
Uhh... That's not ideal indeed, but yeah, they'd have to be public.